modules/standard/GPU.chpl

```diff
@@ -895,6 +895,55 @@ module GPU
 
   private import Time;
 
+  // We no doc it so we can test this independently
+  @chpldoc.nodoc
+  proc gpuCubSort(ref gpuInputArr : [] ?t) {
+    param cTypeName = if      t==int(8)   then "int8_t"
+                      else if t==int(16)  then "int16_t"
+                      else if t==int(32)  then "int32_t"
+                      else if t==int(64)  then "int64_t"
+                      else if t==uint(8)  then "uint8_t"
+                      else if t==uint(16) then "uint16_t"
+                      else if t==uint(32) then "uint32_t"
+                      else if t==uint(64) then "uint64_t"
+                      else if t==real(32) then "float"
+                      else if t==real(64) then "double"
+                      else                     "unknown";
+
+    if cTypeName == "unknown" {
+      compilerError("Arrays with ", t:string,
+                    " elements cannot be sorted with gpuCubSort functions");
+    }
+
+    // Only useful when calling gpuCubSort directly
+    extern proc chpl_gpu_can_cub_sort(): bool;
```

Comment:
Let's not put `cub` in the name. I see that you probably did that because GPU can sort with or without CUB because of the in-house implementation we have. But I don't think that holds much water for me. The api name could change down the road for NVIDIA and AMD. When we have support for Intel, whose software stack is not CUDA-inspired I am sure `cub` will not make much sense.

Reply:
Yeah I put CUB in the name because the function tells us whether we can use CUB for the sort or not. 
I can rename this to `chpl_gpu_can_extern_sort` and rename `gpuCubSort` to gpuExternSort` as well if you'd like.

Reply:
I recommend just `chpl_gpu_can_sort`. It is a runtime function and I view it as asking the runtime whether it can sort it.

I can tolerate `chpl_gpu_can_extern_sort`, if you wanna argue for it more strongly. My reasons for not liking it:

- Doesn't add much in terms of meaning in my view
- Is Longer
- Is different than `chpl_gpu_can_reduce` (obviously, if you want to stick with `chpl_gpu_can_extern_sort`, you should rename the reduce support functions, too)